### PR TITLE
WIP: Updates to Storybook

### PR DESCRIPTION
**Pull request by JohnAlbin**
WIP
 
Currently:
- Removes storybook/yarn.lock so projects can manage it

Problems that still need fixing:
- [x] **MAJOR**: Twig decorator in config breaks when a storiesOf() uses a decorator.
- [x] `twig.tsx` needs to be refactored into a proper Storybook Decorator
- [x] twig.js is included in theme's bundle; move inclusion to storybook config
- [x] Use twig-drupal-filters to add Drupal's Twig filters/functions to storybook
- [x] Twig files should use `import` instead of `require()`
- [ ] Can't run yarn commands from outside storybook folder, e.g. you can't run them from the Drupal theme directory. (too many package.json files)
- [ ] too many webpack.config.js files; no sharing of config. Also need to investigate webpack 4's zero config option
- [x] `stories/.gitkeep` file no longer needed
- [x] Using `Page` as an example story from Silverback means that the main HTML container component in a project will be blown away any time we update Silverback with `silverback init`
- [ ] Does not contain Jest testing
- [x] Usage of random data from `faker` makes Jest testing impossible
- [x] Only works with TypeScript and not regular JS files
- [x] npm shouldn't be installed in node_modules
- [ ] eslint testing is needed
- [ ] eslint and/or prettier formatting is needed

### CODE REVIEW

File: assets/storybook/.storybook/config.js [hunk 1]
```diff
@@ -1,25 +1,20 @@
...
+
+import Twig from 'twig';
+import twigDrupal from 'twig-drupal-filters';
+
+// Add the filters to Drupal.
+twigDrupal(Twig);
 
 // Automatically import all files ending in *.stories.js
-const req = require.context('../stories', true, /\.stories\.tsx?$/);
+const req = require.context('../stories', true, /\.stories\.(ts|js)?$/);
```
Comment (JohnAlbin) [inline assets/storybook/.storybook/config.js]: You forgot to remove the `?`. `/\.stories\.(ts|js)?$/)` should be `/\.stories\.(ts|js)$/)`

File: assets/storybook/scripts.ts [hunk 2]
```diff
@@ -1,11 +1,7 @@
...
 
-import './styles.css';
-
-import Twig from 'twig';
-
-// Fake translation filter, to mock Drupal's behavior.
-Twig.extendFilter("t", function(value) {
-    return value;
-});
+const components = require.context('./stories/twig', true, /\.component\.(ts|js)?$/);
```
Comment (JohnAlbin) [inline assets/storybook/scripts.ts]: Ditto the regex above.

File: assets/storybook/webpack.config.js [hunk 3]
```diff
@@ -3,15 +3,17 @@ const path = require('path');
 const MiniCssExtractPlugin = require("mini-css-extract-plugin");
 
 const jsRule = {
-  test: /\.tsx?$/,
+  test: /\.(ts|js)?$/,
```
Comment (JohnAlbin) [inline assets/storybook/webpack.config.js]: Ditto.

File: assets/storybook/.storybook/webpack.config.js [hunk 4]
```diff
@@ -2,15 +2,17 @@ const path = require('path');
 
 module.exports = ({ config, mode }) => {
 	config.module.rules.push({
-		test: /\.tsx?$/,
+		test: /\.(ts|js)?$/,
```
Comment (JohnAlbin) [inline assets/storybook/.storybook/webpack.config.js]: Ditto.

File: assets/storybook/editor.ts [hunk 5]
```diff
@@ -1 +1,2 @@
-import './stories/editor/columns/columns';
+const req = require.context('./stories/editor', true, /\.component\.(ts|js)?$/);
```
Comment (JohnAlbin) [inline assets/storybook/editor.ts]: Ditto.

File: assets/storybook/.storybook/webpack.config.js [hunk 6]
```diff
@@ -28,7 +30,7 @@ module.exports = ({ config, mode }) => {
 				options: {
 					twigOptions: {
 						namespaces: {
-							storybook: path.resolve('stories')
+							storybook: path.resolve('stories/twig')
```
Comment (JohnAlbin) [inline assets/storybook/.storybook/webpack.config.js]: Can we flatten the directory structure?

Current is:
```
/storybook
/storybook/stories
/storybook/stories/editor
/storybook/stories/twig
```

What about…?
```
/storybook
/storybook/editor
/storybook/components
```

It's easy to have multiple folder sources for Storybook; we had that in UH.

Ex:
```js
// Automatically import all files ending in *.stories.js
const reqEditor = require.context('../editor', true, /\.stories\.(ts|js)$/);
const reqComponents = require.context('../components', true, /\.stories\.(ts|js)$/);

function loadStories() {
  reqEditor.keys().forEach(reqEditor);
  reqComponents.keys().forEach(reqComponents);
}

configure(loadStories, module);
```

Comment (pmelab) [reply assets/storybook/.storybook/webpack.config.js]: Yes, that's fine by me.

**Comment by JohnAlbin** [comment]
👍

**Pull request merged**